main.go

```diff
@@ -88,108 +96,175 @@ func (c *DownloadPlugin) Run(cliConnection plugin.CliConnection, args []string)
 		os.Exit(1)
 	}
 
-	flagVals := ParseFlags(args)
+	cmdExec := cmd_exec.NewCmdExec()
+
+	// parse flags and get list of download paths
+	flagVals, paths := ParseArgs(args)
+	paths = ExpandGlobs(cmdExec, paths, flagVals.Instance_flag)
 
-	// flag variables
-	filterList := filter.GetFilterList(flagVals.Omit_flag, flagVals.Verbose_flag) // get list of things to not download
+	// get list of things to not download
+	filterList := filter.GetFilterList(flagVals.Omit_flag, flagVals.Verbose_flag)
 
+	// get server path to download from and local path to download to
 	workingDir, err := os.Getwd()
 	check(err, "Called by: Getwd")
-	rootWorkingDirectoryServer, startingPathServer := GetDirectoryContext(workingDir, args)
-	rootWorkingDirectoryLocal := filepath.FromSlash(rootWorkingDirectoryServer)
+	pathVals := GetDirectoryContext(workingDir, paths, flagVals.File_flag)
+	for _, v := range pathVals {
+		v.RootWorkingDirectoryLocal = filepath.FromSlash(v.RootWorkingDirectoryLocal)
```

Comment:
If I understand this line correctly, it postprocesses all of the RootWorkingDirectoryLocal values to make them OS-agnostic. Is there a reason that this isn't done inside of `GetDirectoryContext`? It seems like you probably don't want your helper functions emitting OS-unsafe paths, especially given that `GetDirectoryContext` is visible outside of the module.